src/ray/protobuf/core_worker.proto

```diff
@@ -413,12 +413,15 @@ message ReportGeneratorItemReturnsRequest {
 }
 
 message ReportGeneratorItemReturnsReply {
-  // The total number objects consumed from the generator.
-  // -1 means it is not known. The executor side should just
-  // assume everything is consumed if it is -1.
+  // The total number of objects that have been consumed from the generator,
+  // i.e. a ref was returned to the caller of the generator task.  -1 means
+  // unknown and may be returned if an error occurred when requesting from the
+  // caller. The executor side should just assume everything is consumed if it
+  // is -1.
```

Comment:
@rkooo567 is this right? [This code](https://github.com/ray-project/ray/blob/master/src/ray/core_worker/generator_waiter.cc#L61) seems to suggest otherwise.